networkit/cpp/io/MTXGraphReader.cpp

```diff
@@ -0,0 +1,38 @@
+#include <networkit/auxiliary/Enforce.hpp>
+#include <networkit/auxiliary/Log.hpp>
+#include <networkit/auxiliary/StringTools.hpp>
+#include <networkit/io/MTXGraphReader.hpp>
+#include <networkit/io/MTXParser.hpp>
+
+namespace NetworKit {
+
+Graph MTXGraphReader::read(const std::string &path) {
+    MTXParser parser(path);
+
+    auto header = parser.getHeader();
+    auto size = parser.getMatrixSize();
+    auto weighted = true;
+    auto symmetric = true;
+
+    if (header.field == MTXParser::Field::Pattern)
+        weighted = false;
+    if (header.symmetry == MTXParser::Symmetry::General)
+        symmetric = false;
+
+    Graph G(std::max(size.rows, size.columns), weighted, !symmetric);
+    std::string graphName =
+        Aux::StringTools::split(Aux::StringTools::split(path, '/').back(), '.').front();
+
+    std::optional<MTXParser::Edge> current_edge = parser.getNext(weighted);
+
+    while (current_edge.has_value()) {
+        const auto [from, to, weight] = current_edge.value();
+        std::cout<<"from= "<<from<<" to= "<<to<<" weight= "<<*weight<<"\n";
```

Comment:
Based on debugging on MSVC 14.40: The problem is the deref on optional `weight` (given the graph is not weighted). The optional is not default initialized. Instead of fixing it, I suggest use the existing `Graph::Edge` and `Graph::WeightedEdge` together with a template parameter instead of redefining these structs.

The std::cout lines should be removed in the final code anyways.